_posts/2023-04-17-corewcf-1_4_0_preview.markdown

```diff
@@ -0,0 +1,145 @@
+});
+app.Run();
+```
+
+#### RabbitMQ
+The RabbitMQ support has separate service and client packages, CoreWCF.RabbitMQ and CoreWCF.RabbitMQ.Client respectively. While the CoreWCF.RabbitMQ service package supports netstandard2.0, the client only CoreWCF.RabbitMQ.Cient packge supports .NET 6.0 and later. This is due to requiring the latest (as of writing still in pre-release) WCF client packages which support .NET 6.0 or later. These packages will come out of preview once we've had enough feedback/usage to have confidence that we have the correct api's for configuring and developers are able to use these packages successfully.  
+Here's the code you need to use the service binding.
+
+```
+var builder = WebApplication.CreateBuilder();
+builder.Services.AddServiceModelServices()
+                .AddQueueTransport()
+
+var app = builder.Build();
+
+app.UseServiceModel(serviceBuilder =>
+{
+    var uri = new Uri("net.amqps://localhost:5672/amq.direct/corewcf-classic-queue#corewcf-classic-key");
+    var sslOption = new SslOption
+    {
+        ServerName = uri.Host,
+        Enabled = true
+    };
+	// Replace with actual credentials for connecting to RabbitMQ host
+	var credentials = new NetworkCredential(ConnectionFactory.DefaultUser, ConnectionFactory.DefaultPass);
+    serviceBuilder.AddService<Service>();
+    services.AddServiceEndpoint<Service, IService>(
+        new RabbitMqBinding
+        {
+            SslOption = sslOption,
+            Credentials = credentials,
+            QueueConfiguration = new ClassicQueueConfiguration()
+        },
+        uri);
+    });
+});
+app.Run();
+```
+For the queue configuration, we have two types, `ClassicQueueConfiguration` and `QuorumQueueConfiguration`. This configures whether the queue is either a classic mirrored queue, or the newer quorum queue. More information can [be found here](https://www.rabbitmq.com/quorum-queues.html). There are properties on these classes to configure various aspects of the queue such as whether it is durable.  
+There is a mapping of the Uri passed to the endpoint and the queue options. The format of the Uri is `SCHEME://HOSTNAME:PORT/EXCHANGE/QUEUE_NAME#ROUTING_KEY`. The scheme can be one of two values, `net.amqp` or `net.amqps`, with the latter using a TLS secured connection to the AMQP server. The exchange and routing key are optional. If you aren't using an exchange and only have a queue name, ensure the Uri does NOT end in a slash. For example, the Uri `net.amqp://server/queuename` will connect to the queue called queuename and won't use an exchange. You can provide an optional routing key as a Uri fragment.  
+
+For the code example above, we're connecting to an AMQP server running on port 5672, not using TLS (so communication is in the clear). The exchange is `amq.direct`, and the queue name is `corewcf-classic-queue`. It's using the routing key `corewcf-classic-key`.  
```

Comment:
The example above is using TLS (scheme is `net.amqps` and `sslOption.Enabled = true`) which contradicts the text description. Should we reword this or use a different example?

Reply:
I copied this from a test which which using SslOptions and net.amqp which is contradictory. I tried to fix up the sample to be using SSL as that's best practice, but forgot to update the text. Is anything needed on the client side? Should I be setting SslOptions there?